### PR TITLE
Disable the ConnectCallback_UseNamedPipe_Success test for NativeAOT

### PR DESCRIPTION
**Pull request by MihaZupan**
#73772

### CODE REVIEW

File: src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs [hunk 1]
```diff
@@ -3099,6 +3099,12 @@ public sealed class SocketsHttpHandlerTest_ConnectCallback_Http2 : SocketsHttpHa
         [InlineData(false)]
         public async Task ConnectCallback_UseNamedPipe_Success(bool useSsl)
         {
+            if (PlatformDetection.IsWindows && PlatformDetection.IsNativeAot)
```
Comment (CarnaViire) [inline src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs:3102]: Why not an attribute?

Comment (CarnaViire) [reply src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs:3102]: or it is not possible to have an "&&" check in the attribute?

**Review by CarnaViire — APPROVED** [review]
Thanks!

**Comment by danespinosa** [comment]
Thanks for the fix @MihaZupan  and for the help @CarnaViire . I will do some investigation on my end to try to avoid this. My sincere apologies for the inconvenience.

File: src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs (hunk 1 above)
Comment (MihaZupan) [reply src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs:3102]: It's not.
The alternative we've been using is to either add the check into the method with a comment like I did or to introduce a new property that does the AND.

Personally I find the in-method approach cleaner if you're not reusing the same aggregate condition across multiple tests.

Comment (jkotas) [reply src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs:3102]: > or it is not possible to have an "&&"-meaning check in the attribute?

You can use multiple PlatformDetection conditions with the ActiveIssue and other similar attributes. xunit runner will && all of them. 

This should do the right thing here:

`[ActiveIssue("https://github.com/dotnet/runtime/issues/73772", typeof(PlatformDetection), nameof(PlatformDetection.IsWindows), nameof(PlatformDetection.IsNativeAot))]`

Comment (MihaZupan) [reply src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs]: Thanks Jan, that's good to know. Changed it to use the Attribute instead.

**Review by MichalStrehovsky — APPROVED** [review]

**Pull request merged**